contrib/qpl-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,365 @@
+## The Intel® QPL provides high performance implementations of data processing functions for existing hardware accelerator, and/or software path in case if hardware accelerator is not available.
+
+set (QPL_PROJECT_DIR "${ClickHouse_SOURCE_DIR}/contrib/qpl")
+set (QPL_SRC_DIR "${ClickHouse_SOURCE_DIR}/contrib/qpl/sources")
+set (QPL_BINARY_DIR "${ClickHouse_SOURCE_DIR}/build/contrib/qpl")
+set (UUID_DIR "${ClickHouse_SOURCE_DIR}/contrib/qpl-cmake")
+
+set (EFFICIENT_WAIT ON)
+set (BLOCK_ON_FAULT ON)
+set (LOG_HW_INIT OFF)
+set (SANITIZE_MEMORY OFF)
+set (SANITIZE_THREADS OFF)
+set (LIB_FUZZING_ENGINE OFF)
+
+function(GetLibraryVersion _content _outputVar)
+    string(REGEX MATCHALL ".*Qpl VERSION+.* (.+).*LANGUAGES" VERSION_REGEX "${_content}")
+    SET(${_outputVar} ${CMAKE_MATCH_1} PARENT_SCOPE)
+endfunction()
+
+FILE(READ "${QPL_PROJECT_DIR}/CMakeLists.txt" HEADER_CONTENT)
+GetLibraryVersion("${HEADER_CONTENT}" QPL_VERSION)
+
+message(STATUS "Intel QPL version: ${QPL_VERSION}")
+
+include("${QPL_PROJECT_DIR}/cmake/CompileOptions.cmake")
+
+# check nasm compiler
+include(CheckLanguage)
+check_language(ASM_NASM)
+if(NOT CMAKE_ASM_NASM_COMPILER)
+  set (NASM_SOURCE_DIR "${ClickHouse_SOURCE_DIR}/contrib/nasm")
+  set (NASM_BINARY_DIR "${ClickHouse_SOURCE_DIR}/build/contrib/nasm")
+  message(WARNING "Trying to build NASM from ${NASM_SOURCE_DIR} because NASM could not be found on local machine")
+
+  execute_process(COMMAND
+      bash -c "./autogen.sh && ./configure --prefix=${NASM_BINARY_DIR} && make install"
```

Comment:
We should not use `configure` from inside CMake.

Reply:
@alexey-milovidov ASM_NASM compiler is required for this patch (qpl library), how can I enable it in clickhouse source build? do you have more suggestion?
Here I want to compile nasm during cmake configuration, then the qpl library can be compiled in source build.
The nasm official site: https://www.nasm.us/

Reply:
Add it to the Docker image for builds, see `docker/...`

Reply:
But there is a chance that llvm-as has support for Intel syntax, so we don't need nasm to assemble it.

Reply:
> 
@alexey-milovidov There are 3 aspects that I think we can't instead nasm with llvm to do this.
1. Intel syntax is not exactly same as NASM, LLVM support the Intel syntax but not support the Netwide syntax (That's what the 'N' in "NASM' stands for), refer to this link: https://groups.google.com/g/llvm-dev/c/FQA4ET0e8-M 
2. I tried to compile asm target in qpl library with "clang -o xxx.o xxx.asm -masm=intel" but a lot of error happen.
3. As i know so far, llvm don't support avx512 very well , but avx512 is required for asm source build in qpl library.

Reply:
> Add it to the Docker image for builds, see `docker/...`
@alexey-milovidov I've added nasm into docker. But i think this only works on CI setup with docker, for other setup without docker, how can we handle it?

Reply:
@alexey-milovidov could u please approve to run workflows?